[feat] Allow opening workflows with same name as new tabs

Myestery opened the pull request:
When importing a workflow file that has the same name as an existing open workflow, create a new tab with a unique suffix (2), (3), etc. instead of silently failing or replacing the existing workflow.

- Add forceNew parameter to createTemporary() in workflowStore
- Always create new temporary workflow when importing via file picker
- Remove logic that looked up persisted workflows by name during import



https://github.com/user-attachments/assets/9c9aebd3-37c2-464f-9fb4-9ef871ec2673





┆Issue is synchronized with this [Notion page](https://www.notion.so/PR-7104-feat-Allow-opening-workflows-with-same-name-as-new-tabs-2bd6d73d365081cd9182eaa35bf37c29) by [Unito](https://www.unito.io)

Myestery committed:
[feat] Allow opening workflows with same name as new tabs

When importing a workflow file that has the same name as an existing
open workflow, create a new tab with a unique suffix (2), (3), etc.
instead of silently failing or replacing the existing workflow.

- Add forceNew parameter to createTemporary() in workflowStore
- Always create new temporary workflow when importing via file picker
- Remove logic that looked up persisted workflows by name during import

Myestery committed:
[feat] Allow opening workflows with same name as new tabs

When importing a workflow file that has the same name as an existing
open workflow, create a new tab with a unique suffix (2), (3), etc.
instead of silently failing or replacing the existing workflow.

- Add forceNew parameter to createTemporary() in workflowStore
- Always create new temporary workflow when importing via file picker
- Remove logic that looked up persisted workflows by name during import

Myestery committed:
refactor: change ComfyWorkflow import from type-only to regular

DrJKL commented on src/platform/workflow/core/services/workflowService.ts:
```diff
@@ -310,31 +310,37 @@ export const useWorkflowService = () => {
         path ? appendJsonExt(path) : undefined,
-        workflowData
+        workflowData,
+        true
```
Nit: Bare boolean arguments are a code smell, could you convert this to have named parameters / options arg?

Myestery committed:
refactor: split createTemporary into explicit methods to avoid boolean parameter

Replace boolean parameter with two distinct methods (createTemporary and createNewTemporary)
to make intent explicit and improve code readability. This addresses code review feedback
about avoiding bare boolean arguments as function parameters.

Myestery committed:
Merge branch 'main' into allow-multiple-same-workflow-names

Myestery committed:
refactor: clarify workflow reuse logic for loaded files not in directory and use strict equality

christian-byrne reviewed (approved):
LGTM. I thought of some other edge cases, but after testing manually, it seems they all still work properly.
